pkg/ebpf/c/oom-kill-kern.c

```diff
@@ -0,0 +1,44 @@
+#define KBUILD_MODNAME "foo"
+#include <linux/oom.h>
+
+#include "bpf-common.h"
+#include "oom-kill-kern-user.h"
+
+
+/*
+ * The `oomStats` hash map is used to share with the userland program system-probe
+ * the statistics per pid
+ */
+BPF_HASH(oomStats, u32, struct oom_stats);
+
+static inline __attribute__((always_inline))
+bool is_memcg_oom(struct oom_control *oc)
+{
+  return oc->memcg != NULL;
+}
+
+int kprobe__oom_kill_process(struct pt_regs *ctx, struct oom_control *oc, const char *message) {
+    struct oom_stats zero = {};
+    u32 pid = bpf_get_current_pid_tgid() >> 32;
+
+    struct oom_stats *s = oomStats.lookup_or_init(&pid, &zero);
+    if (s == NULL) return 0;
+
+    // From bpf-common.h
+    get_cgroup_name(s->cgroup_name, sizeof(s->cgroup_name));
```

Comment:
Should you be looking at the return value for this at all?

Reply:
it's expected in some cases that there be no cgroup name; but is there a more correct way to handle this? i'm not too familiar with c so open to all feedback!

Reply:
Ah that makes sense - I didn't realize it was optional.